packages/flutter/lib/src/material/toggleable.dart

```diff
@@ -40,22 +43,99 @@ abstract class RenderToggleable extends RenderConstrainedBox implements Semantic
       ..onTap = _handleTap
       ..onTapUp = _handleTapUp
       ..onTapCancel = _handleTapCancel;
+    this.tickerProvider = tickerProvider; // prepares the animation controllers
+  }
+
+  /// Used by subclasses to manipulate the visual value of the control.
+  ///
+  /// Some controls respond to user input by updating their visual value. For
+  /// example, the thumb of a switch moves from one position to another when
+  /// dragged. These controls manipulate this animation controller to update
+  /// their [position] and eventually trigger an [onChanged] callback when the
+  /// animation reaches either 0.0 or 1.0.
+  AnimationController get positionController => _positionController;
+  AnimationController _positionController;
 
+  /// The visual value of the control.
+  ///
+  /// When the control is inactive, the [value] is false and this animation has
+  /// the value 0.0. When the control is active, the value is [true] and this
+  /// animation has the value 1.0. When the control is changing from inactive
+  /// to active (or vice versa), [value] is the target value and this animation
+  /// gradually updates from 0.0 to 1.0 (or vice versa).
+  CurvedAnimation get position => _position;
+  CurvedAnimation _position;
+
+  /// Used by subclasses to control the radial reaction animation.
+  ///
+  /// Some controls have a radial ink reaction to user input. This animation
+  /// controller can be used to start or stop these ink reactions.
+  ///
+  /// Subclasses should call [paintRadialReaction] to actually paint the radial
+  /// reaction.
+  AnimationController get reactionController => _reactionController;
+  AnimationController _reactionController;
+  Animation<double> _reaction;
+
+  /// The [TickerProvider] for the [AnimationController]s that run the animations.
+  TickerProvider get tickerProvider => _tickerProvider;
+  TickerProvider _tickerProvider;
+  set tickerProvider(TickerProvider newTickerProvider) {
+    assert(newTickerProvider != null);
+    if (newTickerProvider == _tickerProvider)
+      return;
+    _tickerProvider = newTickerProvider;
+
+    AnimationController oldController;
+
+    // POSITION CONTROLLER
+    oldController = _positionController;
     _positionController = new AnimationController(
```

Comment:
The problem with this approach is that these animations are exposed publicly by this class.  Clients won't expect them to change.  Specifically, how will they know to move their child animations onto the new instances?

We could solve that by introducing proxy animations to hide the identity change, but I think it would be better to teach AnimationController how to switch ticker providers, especially because of the frame-repeating problem mentioned in another comment.